libraries/AC_Fence/AC_Fence.cpp

```diff
@@ -550,6 +573,74 @@ bool AC_Fence::pre_arm_check(char *failure_msg, const uint8_t failure_msg_len) c
     return true;
 }
 
+/*
+  get our altitude in the FENCE_ALT_TYPE frame
+  return false if not available
+ */
+bool AC_Fence::get_alt_U_in_frame(float &alt, Location::AltFrame alt_frame) const
+{
+    const auto &ahrs = AP::ahrs();
+
+    ahrs.get_relative_position_D_home(alt);
+
+    // translate to +ve up
+    alt = -alt;
+
+    /*
+      convert to desired altitude frame
+     */
+    switch (alt_frame) {
```

Comment:
Should there not also be a switch case for 2: Above Origin (Location::AltFrame::ABOVE_ORIGIN)?